### DISCUSSION TITLE
How to prevent student from editing GitHub Autograding Assignment Setup?

### DISCUSSION THREAD

**Discussion by enamcse** [community/community#17938] [GitHub Education]
As of now, I know the following ways to set up an auto-graded assignment in GitHub.

1. Creating a bash file to check tests. Run the bash file with GitHub actions. The setup for this GitHub Action/Workflow would be in a folder called `.github` in the repository.
2. Creating tests using GitHub Actions directly. In that case, The setup files remain in the `.github` folder like before.
3.  There might be some other ways where a third-party app's intervention is needed.

It seems like the discussion came into the community near April 2020. [Here](https://education.github.community/t/prevent-students-from-changing-autograding-tests/53718) is the detailed discussion.

There, one of the GitHub Staff said that we would not prevent student from editing but we can detect if a student change that. 

I want to know the update and/or I am interested in if any other easy way to prevent/detect this cheating.

Grant my sincere apology if the question is too noob or if I missed something or if I hurt someone.

**Comment by agilebotanist** [comment]
Hi I'm new to Classroom, but have few ideas on a solution that I'd be happy to discuss.

We are creating labs for 300+ students for an advanced course on software quality and CI with GitHub Actions.

We chose to try GitHub Classroom with Authograding in Python. 

We immediately noticed that Autograding action could be easily diverted by students who know a bit about GitHub Actions. If you change a few settings, all assignments pass by magic. It is also possible to add extra points. 

I'm looking for methods to protect auto-grading settings and obfuscate certain tests so that students cannot guess solutions by just copying our tests.

What are the best practices?

So far, the only possible solution that we have found is to download all repositories after the deadline and overwrite the settings files in post-processing. Are there any examples that we can use as inspiration?

**Reply by jwood36** [reply]
I am looking at a solution that uses a separate Private Repo to hold "hidden" unit tests.

I totally expect the students to "cheat" and edit the included unit tests; therefore, the "hidden" unit tests will include those tests, plus additional edge or situational tests..
